src/middlewared/middlewared/plugins/gluster_linux/volume.py

```diff
@@ -0,0 +1,542 @@
+from gfs.cli import volume
+from gfs.cli.utils import GlusterCmdException
+
+from middlewared.service import (CRUDService, accepts,
+                                 job, private, ValidationErrors,
+                                 CallError)
+from middlewared.schema import Dict, Str, Int, Bool, List
+
+from .utils import validate_gluster_jobs
+
+
+class GlusterVolumeService(CRUDService):
+
+    class Config:
+        namespace = 'gluster.volume'
+
+    def __volume_wrapper(self, method, *args, **kwargs):
+
+        result = b''
+
+        try:
+            result = method(*args, **kwargs)
+        except GlusterCmdException as e:
+            rc, out, err = e.args[0]
+            err = err if err else out
+            raise CallError(f'{err.decode().strip()}')
+
+        if isinstance(result, bytes):
```

Comment:
This all ends up being run using `subprocess`. I've found, for example, if you try to `start` a gluster volume that has already been started it will not raise an exception, it simply returns `stdout` so I need to decode it at that point.

Reply:
Right, but why the isinstance call? isnt it always supposed to return bytes?

Reply:
Fair enough, thanks!